docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -286,89 +486,258 @@ is already imported and used in `ContentView.swift` with `import shared`.
     * `ContentView.ViewModel` is declared as an `ObservableObject`.
-    * The `@Published` wrapper is used for the `text` property.
+    * The `@Published` wrapper is used for the `greetings` property.
     * The `@ObservedObject` property wrapper is used to subscribe to the view model.
 
-   Now the view model will emit signals whenever this property changes.
+Now, the view model will emit signals whenever this property changes.
 
-4. Call the `greet()` function, which now also loads data from the SpaceX API, and save the result in the `phrases` property:
+#### Choose a library to consume flows from iOS
 
-    ```swift
-    class ViewModel: ObservableObject {
-        @Published var phrases: [String] = ["Loading..."]
-        init() {
-            Greeting().greet { greeting, error in
-                DispatchQueue.main.async {
-                    if let greeting = greeting {
-                        self.phrases = greeting
-                    } else {
-                        self.phrases = [error?.localizedDescription ?? "error"]
-                    }
-                }
-            }
+In this tutorial, you can choose between the [KMP-NativeCoroutines](https://github.com/rickclephas/KMP-NativeCoroutines)
+and [SKIE](https://github.com/touchlab/SKIE/) libraries that help work with flows in iOS. Both are open-sourced solutions
+that support cancellation and generics with flows, something that the Kotlin/Native compiler doesn't provide by default yet.
+
+The SKIE library augments the Objective-C API produced by the Kotlin Multiplatform compiler.
+SKIE transforms flows into an equivalent of Swift's `AsyncSequence` structure.
+Rather than using a wrapper, it behaves like any other Swift library when calling it from Swift.
+
+Using SKIE is less verbose as it doesn't use wrappers around flows. It's easier to set up because it will be part of the
+framework rather than a dependency that needs to be added as a CocoaPod or SPM file. Apart from flows, SKIE also supports
+[other features](https://skie.touchlab.co/features/).
+
+The KMP-NativeCoroutines library helps you consume suspending functions and flows from iOS. It's a more tried-and-tested
+option and maybe a more stable solution at the moment. KMP-NativeCoroutines supports `async`/`await`, Combine,
+and RxSwift approaches to concurrency, while SKIE supports `async`/`await` only.
+
+#### Option 1. Configure KMP-NativeCoroutines {initial-collapse-state="collapsed"}
+
+> We recommend using the latest version of the library.
+> Check the [KMP-NativeCoroutines repository](https://github.com/rickclephas/KMP-NativeCoroutines/releases) to see if a newer version of the plugin is available.
+>
+{type="note"}
+
+1. Get back to Android Studio. In the `build.gradle.kts` file of the _whole project_,
+   add the KSP (Kotlin Symbol Processor) and KMP-NativeCoroutines plugins to the `plugins` section:
```

Comment:
```suggestion
   add the KSP (Kotlin Symbol Processor) and KMP-NativeCoroutines plugins to the `plugins {}` block:
```

Reply:
But we use them :)
Moreover, it's the official Gradle way how to use them in the docs.